printer_tray/report_xml_action.py

```diff
@@ -19,20 +19,21 @@
 #
 ##############################################################################
 
-from openerp.osv import orm, fields
+from openerp import models, fields, api
 
 
-class ReportXMLAction(orm.Model):
+class ReportXMLAction(models.Model):
     _inherit = 'printing.report.xml.action'
 
-    _columns = {
-        'printer_tray_id': fields.many2one(
-            'printing.tray', 'Paper Source',
-            domain="[('printer_id', '=', printer_id)]"),
-        }
+    printer_tray_id = fields.Many2one(
```

Comment:
You should put an onchange on printer to select the first available tray from new printer